### PR TITLE
fix: [M3-8274] - Fix parsing issue causing in Kubernetes Version field

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝
This PR handles decimal value correctly in Kubernetes Version field (LKE Create Page).

## Target release date 🗓️
6/24

| Before  | After   |
| ------- | ------- |
|image |image |

### Reproduction steps
(How to reproduce the issue, if applicable)
- Navigate to LKE create page in dev
- Notice latest version of Kubernetes Version field. - missing decimal number

### Verification steps
(How to verify changes)
- Checkout this branch
- Navigate to LKE create page
- Notice latest version of Kubernetes Version field. - renders decimal number correctly

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Kubernetes/kubeUtils.ts [hunk 1]
```diff
@@ -119,7 +119,9 @@ export const getLatestVersion = (
   for (const element of versions) {
     versionsNumbersArray.push(parseFloat(element.value));
   }
-  const latestVersionValue = Math.max.apply(null, versionsNumbersArray);
+  const latestVersionValue = Math.max
+    .apply(null, versionsNumbersArray)
+    .toFixed(2);
```
Comment (jdamore-linode) [inline packages/manager/src/features/Kubernetes/kubeUtils.ts:124]: I don't think this fixes the root of the issue. For example, if we had an LKE version `1.5` and an LKE version `1.15`, this function would return an object with `version: '1.50'`.

We probably shouldn't be treating these strings as floats at all. We should be comparing each component of the version string instead.

Comment (cpathipa) [reply packages/manager/src/features/Kubernetes/kubeUtils.ts]: Thank you @jdamore-linode

File: packages/manager/src/utilities/sort-by.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  it('should identify the later minor version as greater', () => {
+    const result = sortByVersion('1.2.0', '1.1.0', 'asc');
+    expect(result).toBeGreaterThan(0);
+  });
+
+  it('should identify the later patch version as greater', () => {
+    const result = sortByVersion('1.1.2', '1.1.1', 'asc');
+    expect(result).toBeGreaterThan(0);
+  });
```
Comment (jdamore-linode) [inline packages/manager/src/utilities/sort-by.test.ts:17]: Thanks again for the tests, these are great!

```suggestion
  });
  
  it('should identify the later minor version with differing number of digits', () => {
    const result = sortByVersion('1.30', '1.3', 'asc');
    expect(result).toBeGreaterThan(0);
  });
```

Maybe one extra test just to cover the "1.3" vs "1.30" scenario that led to this ticket

File: packages/manager/src/features/Kubernetes/kubeUtils.test.ts [hunk 3]
```diff
@@ -64,4 +67,35 @@ describe('helper functions', () => {
...
+    it('should handle latest version minor version correctly', () => {
+      const versions = [
+        { label: '1.22', value: '1.22' },
+        { label: '1.23', value: '1.23' },
+        { label: '1.30', value: '1.30' },
+      ];
+      const result = getLatestVersion(versions);
+      expect(result).toEqual({ label: '1.30', value: '1.30' });
+    });
+    it('should handle latest path version correctly', () => {
```
Comment (jdamore-linode) [inline packages/manager/src/features/Kubernetes/kubeUtils.test.ts:90]:
```suggestion
    it('should handle latest patch version correctly', () => {
```

**Review by jdamore-linode — APPROVED** [review]
Nice work @cpathipa, thank you for the extra test coverage!

Was able to reproduce the issue in `develop` and confirmed that on this branch the correct version is being selected 👍

File: packages/manager/src/utilities/sort-by.ts [hunk 4]
```diff
@@ -45,3 +45,25 @@ export const sortByArrayLength = (a: any[], b: any[], order: SortOrder) => {
 
   return order === 'asc' ? result : -result;
 };
+
+export const sortByVersion = (a: string, b: string, order: SortOrder) => {
```
Comment (abailly-akamai) [inline packages/manager/src/utilities/sort-by.ts:49]: Let's add a JSDoc comment and return type

File: packages/manager/src/features/Kubernetes/kubeUtils.ts [hunk 5]
```diff
@@ -114,12 +115,10 @@ export const getKubeHighAvailability = (
...
-    versionsNumbersArray.push(parseFloat(element.value));
-  }
-  const latestVersionValue = Math.max.apply(null, versionsNumbersArray);
+  const sortedVersions = [...versions].sort((a, b) => {
+    return sortByVersion(a.value, b.value, 'asc');
+  });
 
-  return { label: `${latestVersionValue}`, value: `${latestVersionValue}` };
+  const latestVersion = sortedVersions.pop();
+  return { label: `${latestVersion?.value}`, value: `${latestVersion?.value}` };
```
Comment (abailly-akamai) [inline packages/manager/src/features/Kubernetes/kubeUtils.ts:123]: this could end up printing 'undefined' to the page

**Review by abailly-akamai — APPROVED** [review]
Thanks for addressing the changes - e2e failures seem unrelated but another run to confirm that wouldn't hurt

**Comment by cpathipa** [comment]
Failing OBJ tests are not related to this PR. Hence merging it to develop.

**Pull request merged**